cloudinit/sources/DataSourceVMware.py

```diff
@@ -527,6 +669,73 @@ def advertise_local_ip_addrs(host_info, rpctool, rpctool_fn):
         LOG.info("advertised local ipv6 address %s in guestinfo", local_ipv6)
 
 
+def advertise_update_events(
+    supported_update_events, enabled_update_events, rpctool, rpctool_fn
+):
+    """
+    advertise_update_events publishes the types of supported and
+    enabled events to guestinfo.
+
+    The string returned from this method adheres to the following
+    format:
+
+        SCOPE=TYPE[;TYPE][,SCOPE=TYPE[;TYPE]]
+
+    For example:
+
+        * network=boot;hotplug
+        * network=boot-new-instance
+
+    The only supported event scope at the moment is network, but there
+    may be more scopes in the future, and this format will support them.
+    """
+    if not rpctool or not rpctool_fn:
+        return None, None
+
+    def get_events_string(events):
```

Comment:
I think this could be simplified to:
```python
def get_events_string(events):
    event_scopes_and_types_list = []
    for event_scope, event_types in events.items():
        event_types_list = sorted(
            str(event_type) for event_type in event_types
        )
        event_scopes_and_types_list.append(
            "{}={}".format(event_scope, ";".join(event_types_list))
        )
    return ",".join(event_scopes_and_types_list)
```